manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/builder/IstioResourcesManifestGenerator.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.generator.services.builder;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.models.LoadBalancer;
+import io.hyscale.commons.models.ServiceMetadata;
+import io.hyscale.plugin.framework.models.ManifestSnippet;
+
+public interface IstioResourcesManifestGenerator {
```

Comment:
Instead of having this as an interface, can we make it an abstract class and have the common functionality here, the implementation will only have to provide templateType, context, path, and kind. What do you say?